readme: Reestructure readme and add resources link (#23847)
redpanda-data/redpanda: README.md

<a href="https://redpanda.com/redpanda"><img src="docs/PANDA_sitting.png" alt="redpanda sitting" width="400"></a>
</p>

Redpanda is the most complete, Apache Kafka®-compatible streaming data platform, designed from the ground up to be lighter, faster, and simpler to operate. Free from ZooKeeper™ and JVMs, it prioritizes an end-to-end developer experience with a huge ecosystem of connectors, configurable tiered storage, and more.

# Table of Contents

> - [Get started](#getting-started)

Author: ```suggestion
- [Get started](#get-started)
```

  - [Prebuilt packages](#prebuilt-packages)
    - [Debian/Ubuntu](#debianubuntu)
    - [Fedora/RedHat/Amazon Linux](#fedoraredhatamazon-linux)
    - [macOS](#macos)
    - [Other Linux environments](#other-linux-environments)
  - [Build manually](#build-manually)
  - [Release candidate builds](#release-candidate-builds)
    - [RC releases on Debian/Ubuntu](#rc-releases-on-debianubuntu)
    - [RC releases on Fedora/RedHat/Amazon Linux](#rc-releases-on-fedoraredhatamazon-linux)
    - [RC releases on Docker](#rc-releases-on-docker)
- [Community](#community)

> - [Resources](#additional-links-and-resources)

Author: ```suggestion
- [Resources](#resources)
```

<a href="https://redpanda.com/redpanda"><img src="docs/PANDA_sitting.png" alt="redpanda sitting" width="400"></a>
</p>

Redpanda is the most complete, Apache Kafka®-compatible streaming data platform, designed from the ground up to be lighter, faster, and simpler to operate. Free from ZooKeeper™ and JVMs, it prioritizes an end-to-end developer experience with a huge ecosystem of connectors, configurable tiered storage, and more.

# Table of Contents

> - [Redpanda](#redpanda)
> - [Table of Contents](#table-of-contents)

Reviewer: ```suggestion
```

> - [Getting Started](#getting-started)

Reviewer: ```suggestion
- [Get started](#getting-started)
```

>   - [Prebuilt Packages](#prebuilt-packages)

Reviewer: ```suggestion
  - [Prebuilt packages](#prebuilt-packages)
```

    - [Debian/Ubuntu](#debianubuntu)
    - [Fedora/RedHat/Amazon Linux](#fedoraredhatamazon-linux)
    - [macOS](#macos)
    - [Other Linux environments](#other-linux-environments)

>   - [Build Manually](#build-manually)

Reviewer: ```suggestion
  - [Build manually](#build-manually)
```

  - [Release candidate builds](#release-candidate-builds)
    - [RC releases on Debian/Ubuntu](#rc-releases-on-debianubuntu)
    - [RC releases on Fedora/RedHat/Amazon Linux](#rc-releases-on-fedoraredhatamazon-linux)
    - [RC releases on Docker](#rc-releases-on-docker)
- [Community](#community)

> - [Additional Links and resources](#additional-links-and-resources)

Reviewer: ```suggestion
- [Additional links and resources](#additional-links-and-resources)
```

> # Getting Started

Reviewer: ```suggestion
# Get started
```

> ## Prebuilt Packages

Reviewer: ```suggestion
## Prebuilt packages
```

> Redpanda Data recommends using the free & prebuilt stable releases below.

Reviewer: ```suggestion
Redpanda Data recommends using the following free, prebuilt stable releases.
```

sudo yum install redpanda
```

### macOS

> Simply download our `rpk` [binary here](https://github.com/redpanda-data/redpanda/releases). We require Docker on MacOS

Reviewer: ```suggestion
Simply download our `rpk` [binary here](https://github.com/redpanda-data/redpanda/releases). We require Docker on MacOS.
```

```
brew install redpanda-data/tap/redpanda && rpk container start
```

### Other Linux environments

To install from a `.tar.gz` archive, download the file and extract it into `/opt/redpanda`.

For amd64:

```
curl -LO \
  https://dl.redpanda.com/nzc4ZYQK3WRGd9sy/redpanda/raw/names/redpanda-amd64/versions/23.3.6/redpanda-24.2.7-amd64.tar.gz
```

For arm64:

```
curl -LO \
  https://dl.redpanda.com/nzc4ZYQK3WRGd9sy/redpanda/raw/names/redpanda-arm64/versions/23.3.6/redpanda-24.2.7-arm64.tar.gz
```

> Replace `24.2.7` with the appropriate version you are trying to download. See [Release Notes](https://github.com/redpanda-data/redpanda/releases).

Reviewer: ```suggestion
Replace `24.2.7` with the version you want to download. See [Release Notes](https://github.com/redpanda-data/redpanda/releases).
```

> For more build configurations check out `.bazelrc`.

Reviewer: ```suggestion
For more build configurations, see `.bazelrc`.
```

## Release candidate builds

> Redpanda Data create a release candidate (RC) build when we get close to a new release and publish these to make new features available for testing.

Reviewer: ```suggestion
Redpanda Data creates a release candidate (RC) build when we get close to a new release, and we publish it to make new features available for testing.
```

sudo yum install redpanda
```

### RC releases on Docker

> Example with the `v23.1.1-rc1`. A version prior to the 23.1.1 release.

Reviewer: ```suggestion
Example with `v23.1.1-rc1`:
```

Reviewer: should we use something more recent?

Author: Eh, I thought about this, but this will always be a cat-mouse scenario without automation. I don't mind leaving an old version there.

```bash
docker pull docker.redpanda.com/redpandadata/redpanda-unstable:v23.1.1-rc1
```

# Community

> [Slack](https://redpanda.com/slack) is the main way the community interacts with one another in real-time :).

Reviewer: ```suggestion
[Slack](https://redpanda.com/slack) is the main way the community interacts with one another in real-time :).
```
```suggestion
[Slack](https://redpanda.com/slack): This is the main way the community communicates in real-time :).
```

> [Github Discussion](https://github.com/redpanda-data/redpanda/discussions) is preferred for longer, async, thoughtful discussions.

Reviewer: ```suggestion
[Github discussion](https://github.com/redpanda-data/redpanda/discussions)L This is for longer, async, thoughtful discussions.
```

> [GitHub Issues](https://github.com/redpanda-data/redpanda/issues) is reserved only for actual issues. Please use the mailing list for discussions.

Reviewer: ```suggestion
[GitHub issues](https://github.com/redpanda-data/redpanda/issues): This is reserved only for actual issues. Please use the mailing list for discussions.
```

[Code of conduct](./CODE_OF_CONDUCT.md) code of conduct for the community.

> [Contributing to the code](./CONTRIBUTING.md)

Reviewer: ```suggestion
[Contribute to the code](./CONTRIBUTING.md)
```

> # Additional Links and resources

Reviewer: ```suggestion
# Additional links and resources
```

> [Official Redpanda documentation](https://docs.redpanda.com/home/)

Reviewer: ```suggestion
[Redpanda Documentation](https://docs.redpanda.com/home/)
```

> [Official Redpanda blog](https://www.redpanda.com/blog)

Reviewer: ```suggestion
[Redpanda blogs](https://www.redpanda.com/blog)
```

> [Upcoming Redpanda events](https://www.redpanda.com/events)

Reviewer: ```suggestion
[Upcoming Redpanda Events](https://www.redpanda.com/events)
```

> [Redpanda support](https://support.redpanda.com/)

Reviewer: ```suggestion
[Redpanda Support](https://support.redpanda.com/)
```
